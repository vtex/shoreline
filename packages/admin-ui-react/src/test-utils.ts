import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
