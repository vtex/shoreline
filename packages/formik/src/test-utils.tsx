import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
