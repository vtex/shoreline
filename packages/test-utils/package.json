{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
