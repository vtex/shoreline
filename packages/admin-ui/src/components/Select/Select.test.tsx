import React from 'react'
import { render, axe } from '../../test-utils'

import { Select } from './index'

describe('Select', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(
      <Select
        id="select"
        label="Select"
        name="select"
        value="option-1"
        onChange={() => null}
      >
        <option value="option-1">Option 1</option>
        <option value="option-2">Option 2</option>
      </Select>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
