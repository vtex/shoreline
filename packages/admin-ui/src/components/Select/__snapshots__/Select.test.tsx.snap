// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match the snapshot 1`] = `
<DocumentFragment>
  .emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-4 input:focus+label,
.emotion-4 textarea:focus+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-4 input:placeholder-shown:not(:focus)+label,
.emotion-4 textarea:placeholder-shown:not(:focus)+label {
  padding-top: 0.25rem;
}

.emotion-4 input:not(:placeholder-shown)+label,
.emotion-4 textarea:not(:placeholder-shown)+label {
  -webkit-transform: translate(1px, 4px) scale(0.875);
  -moz-transform: translate(1px, 4px) scale(0.875);
  -ms-transform: translate(1px, 4px) scale(0.875);
  transform: translate(1px, 4px) scale(0.875);
}

.emotion-0 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.75rem;
  line-height: 1.5;
  letter-spacing: 0.001em;
  position: absolute;
  top: 0.5rem;
  padding-left: 0.75rem;
  color: hsla(0, 0%, 33%, 1);
  z-index: 2;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
}

.emotion-2 {
  width: 1rem;
  height: 1rem;
  min-height: 1rem;
  min-width: 1rem;
  position: absolute;
  right: 12px;
  top: 25%;
  color: hsla(0, 0%, 33%, 1);
}

.emotion-3 {
  position: relative;
  width: 288px;
  height: 48px;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding-top: 1.125rem;
  padding-left: 0.75rem;
  width: 100%;
  height: 100%;
  border-radius: 4px;
  color: #000000;
  background: transparent;
  border: 1px solid hsla(0, 0%, 73%, 1);
}

.emotion-1:disabled {
  background: hsla(0, 0%, 90%, 1);
  border: 1px solid hsla(0, 0%, 45%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-1:focus {
  border: 1px solid hsla(0, 0%, 23%, 1);
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

<div
    class="emotion-4"
  >
    <div
      class="emotion-3"
    >
      <label
        class="emotion-0"
      >
        Select
      </label>
      <select
        class="emotion-1"
      >
        <option
          value="option-1"
        >
          Option 1
        </option>
        <option
          value="option-2"
        >
          Option 2
        </option>
      </select>
      <svg
        class="emotion-2"
        fill="currentColor"
        height="1rem"
        viewBox="0 0 256 256"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="16rem"
          width="16rem"
        />
        <polyline
          fill="none"
          points="208 96 128 176 48 96"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="18"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
